renderdoc/driver/vulkan/wrappers/vk_misc_funcs.cpp

```diff
@@ -1767,6 +1767,346 @@ VkResult WrappedVulkan::vkGetQueryPoolResults(VkDevice device, VkQueryPool query
+      }
+
+      AddAction(action);
+
+      VulkanActionTreeNode &actionNode = GetActionStack().back()->children.back();
+
+      if(CopyImageToImageInfo.srcImage == CopyImageToImageInfo.dstImage)
+      {
+        actionNode.resourceUsage.push_back(
+            make_rdcpair(GetResID(CopyImageToImageInfo.srcImage),
+                         EventUsage(actionNode.action.eventId, ResourceUsage::Copy)));
+      }
+      else
+      {
+        actionNode.resourceUsage.push_back(
+            make_rdcpair(GetResID(CopyImageToImageInfo.srcImage),
+                         EventUsage(actionNode.action.eventId, ResourceUsage::CopySrc)));
+        actionNode.resourceUsage.push_back(
+            make_rdcpair(GetResID(CopyImageToImageInfo.dstImage),
+                         EventUsage(actionNode.action.eventId, ResourceUsage::CopyDst)));
+      }
+    }
+  }
+
+  return true;
+}
+
+VkResult WrappedVulkan::vkCopyImageToImageEXT(VkDevice device,
+                                              const VkCopyImageToImageInfo *pCopyImageToImageInfo)
+{
+  SCOPED_DBG_SINK();
+
+  byte *tempMem = GetTempMemory(GetNextPatchSize(pCopyImageToImageInfo));
+  VkCopyImageToImageInfo *unwrappedInfo =
+      UnwrapStructAndChain(m_State, tempMem, pCopyImageToImageInfo);
+
+  VkResult ret;
+  SERIALISE_TIME_CALL(ret = ObjDisp(device)->CopyImageToImageEXT(Unwrap(device), unwrappedInfo));
+
+  if(IsActiveCapturing(m_State))
+  {
+    CACHE_THREAD_SERIALISER();
+
+    ser.SetActionChunk();
+    SCOPED_SERIALISE_CHUNK(VulkanChunk::vkCopyImageToImage);
+    Serialise_vkCopyImageToImageEXT(ser, device, pCopyImageToImageInfo);
+
+    m_FrameCaptureRecord->AddChunk(scope.Get());
+    for(uint32_t i = 0; i < pCopyImageToImageInfo->regionCount; i++)
+    {
+      const VkImageCopy2 &region = pCopyImageToImageInfo->pRegions[i];
+
+      ImageRange srcRange(region.srcSubresource);
+      srcRange.offset = region.srcOffset;
+      srcRange.extent = region.extent;
+
+      ImageRange dstRange(region.dstSubresource);
+      dstRange.offset = region.dstOffset;
+      dstRange.extent = region.extent;
+
+      // TODO: vkCmdCopyImage has GetRecord(commandBuffer), what record to use here?
+      // record->MarkImageFrameReferenced(GetRecord(pCopyImageToImageInfo->srcImage), srcRange,
+      // eFrameRef_Read); record->MarkImageFrameReferenced(GetRecord(pCopyImageToImageInfo->dstImage),
+      // dstRange, eFrameRef_CompleteWrite);
+    }
+  }
+
+  return ret;
+}
+
+template <typename SerialiserType>
+bool WrappedVulkan::Serialise_vkCopyImageToMemoryEXT(
+    SerialiserType &ser, VkDevice device, const VkCopyImageToMemoryInfo *pCopyImageToMemoryInfo)
+{
+  SERIALISE_ELEMENT(device);
+  SERIALISE_ELEMENT_LOCAL(CopyImageToMemoryInfo, *pCopyImageToMemoryInfo).Important();
+
+  // The host memory found in pRegions needs to be serialised separately since `this`
+  // (WrappedVulkan) is needed to find its size.
```

Comment:
Again, this is most unfortunate. What I have here works, but it has the side effect that the serialized member is not shown in the right place in the UI:

<img width="450" alt="image" src="https://github.com/user-attachments/assets/52541273-d272-4558-b590-ec64f6a69d8e" />

Is there anything at all that could be done about it?

Reply:
(Resolving per the other comment that says we can move the serialization to vk_serialise.cpp)